### PR TITLE
Feat: survey enter modal

### PR DESCRIPTION
**Pull request by sanghyun1128**
Show summary modals before enter the survey.

Also add useOnClickOutside hook for exit modal when click outside.

### CODE REVIEW

File: web/src/hooks/useOnClickOutside.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      handler();
+    };
+    document.addEventListener('mousedown', listener);
+    return () => {
+      document.removeEventListener('mousedown', listener);
+    };
+  }, [ref, handler]);
+};
+
+export default useOnClickOutside;
```
Comment (seiwonpark) [inline web/src/hooks/useOnClickOutside.ts]: Please export hook with `export const`

File: web/src/components/Modal/SurveyEnterModal.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+  text-decoration: underline;
+  text-decoration-color: currentColor;
+  text-underline-offset: 5px;
+
+  @media screen and (max-width: 960px) {
+    width: 35vw;
+  }
+`;
+
+const Description = styled(Label)`
```
Comment (seiwonpark) [inline web/src/components/Modal/SurveyEnterModal.tsx]: If this is a description, `<p>` tag should match better than `<label>`. `<label>` is for some tags and like category or something.

Comment (seiwonpark) [reply web/src/components/Modal/SurveyEnterModal.tsx]: Or use `<span>`.

**Review by seiwonpark — APPROVED** [review]
Partial LGTM. You can merge this right away or with minor fixes. Thank you!

**Review by ohsung1125 — APPROVED** [review]
LGTM!

**Pull request merged**